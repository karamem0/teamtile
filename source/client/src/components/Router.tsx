//
// Copyright (c) 2023 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { inTeams } from '../utils/Teams';

import Presenter from './Router.presenter';

function Router() {

  return (
    <Presenter inTeams={inTeams()} />
  );

}

export default Router;
