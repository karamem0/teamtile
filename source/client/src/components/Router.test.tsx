//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { render, screen } from '@testing-library/react';

import Router from './Router.presenter';

jest.mock('./AppLoader', () =>
  function AppLoader({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="AppLoader">
        {children}
      </div>
    );
  });

jest.mock('./Snackbar', () =>
  function Snackbar({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="Snackbar">
        {children}
      </div>
    );
  });

jest.mock('../features/auth/pages/CallbackPage', () =>
  function CallbackPage({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="CallbackPage">
        {children}
      </div>
    );
  });

jest.mock('../features/auth/pages/LoginPage', () =>
  function LoginPage({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="LoginPage">
        {children}
      </div>
    );
  });

jest.mock('../features/home/pages/HomePage', () =>
  function HomePage({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="HomePage">
        {children}
      </div>
    );
  });

jest.mock('../features/team/pages/TeamPage', () =>
  function TeamPage({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="TeamPage">
        {children}
      </div>
    );
  });

jest.mock('../pages/NotFoundPage', () =>
  function NotFoundPage({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="NotFoundPage">
        {children}
      </div>
    );
  });

jest.mock('../providers/ReducerProvider', () =>
  function ReducerProvider({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="ReducerProvider">
        {children}
      </div>
    );
  });

jest.mock('../providers/SnackbarProvider', () =>
  function SnackbarProvider({ children }: React.PropsWithChildren<unknown>) {
    return (
      <div data-testid="SnackbarProvider">
        {children}
      </div>
    );
  });

test('create shapshot of when app is in teams', async () => {
  const params = {
    inTeams: true
  };
  render(<Router {...params} />);
  expect((await screen.findAllByText(/^.*$/))[0]).toMatchSnapshot();
});

test('create shapshot of when app is not in teams', async () => {
  const params = {
    inTeams: false
  };
  render(<Router {...params} />);
  expect((await screen.findAllByText(/^.*$/))[0]).toMatchSnapshot();
});
