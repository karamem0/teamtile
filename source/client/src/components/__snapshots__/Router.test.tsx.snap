// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create shapshot of when app is in teams 1`] = `
@media (max-width: 599px) {
  .emotion-0 {
    padding: 0.5rem;
  }
}

@media (min-width: 600px) {
  .emotion-0 {
    padding: 1rem;
  }
}

<body>
  <div>
    <div
      class="emotion-0"
    >
      <div
        data-testid="ReducerProvider"
      >
        <div
          data-testid="SnackbarProvider"
        >
          <div
            data-testid="Snackbar"
          />
          <div
            data-testid="AppLoader"
          >
            <div
              data-testid="TeamPage"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`create shapshot of when app is not in teams 1`] = `
<body>
  <div>
    <div
      data-testid="HomePage"
    />
  </div>
</body>
`;
